datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -955,10 +956,17 @@ public static String getTaskMetadataGroupId(DatastreamTask task, CommonConnector
    */
   private KafkaPositionTracker createKafkaPositionTracker(KafkaBasedConnectorConfig config) {
     if (config.getEnablePositionTracker()) {
-      // Change the consumer group id for our position tracker's consumer as we do not want the position tracker to
-      // be subscribing or assigning itself to any topics
       Properties positionTrackerConsumerProps = new Properties(_consumerProps);
-      positionTrackerConsumerProps.remove(ConsumerConfig.GROUP_ID_CONFIG);
+
+      // Generate a random consumer group id for our position tracker's consumer as we do not want the position tracker
+      // to interfere with the task's consumer (can happen due to KAFKA-8350).
+      String positionTrackerGroupId = UUID.randomUUID().toString();
+      positionTrackerConsumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, positionTrackerGroupId);
+
+      // Increase the request timeout of the position tracker's consumer as it may be asking for a large number of
+      // topics in its RPC calls.
+      int positionTrackerRequestTimeoutMs = (int) Duration.ofMinutes(5).toMillis();
```

Comment:
Could you create a static constant to store the default timeout?

Reply:
Yes, and I will change it to make it user-configurable away from the default.